assets/theme-editor.js

```diff
@@ -26,6 +26,7 @@ document.addEventListener('shopify:block:deselect', function(event) {
 });
 
 document.addEventListener('shopify:section:load', () => {
+  initializeAnimationTrigger();
```

Comment:
Now that I think of it themes can use the [section rendering api](https://shopify.dev/docs/api/section-rendering) on storefronts so we have to watch for this event on storefronts as well, not just the editor. Let's add a `
document.addEventListener('shopify:section:load', () => {` to the scroll trigger code.